src/plugins/data_source/server/client/client_config.ts

```diff
@@ -18,14 +28,70 @@ export function parseClientOptions(
   endpoint: string,
   registeredSchema: any[]
 ): ClientOptions {
+  const sslConfig: DataSourceSSLConfigOptions = {
+    requestCert: true,
+    rejectUnauthorized: true,
+  };
+
+  if (config.ssl) {
+    const verificationMode = config.ssl.verificationMode;
+    switch (verificationMode) {
+      case 'none':
+        sslConfig.rejectUnauthorized = false;
+        break;
+      case 'certificate':
+        sslConfig.rejectUnauthorized = true;
+
+        // by default, NodeJS is checking the server identify
+        sslConfig.checkServerIdentity = () => undefined;
+        break;
+      case 'full':
+        sslConfig.rejectUnauthorized = true;
+        break;
+      default:
+        throw new Error(`Unknown ssl verificationMode: ${verificationMode}`);
+    }
+
+    const { certificateAuthorities } = readCertificateAuthorities(config);
+
+    sslConfig.ca = certificateAuthorities || [];
+  }
+
   const clientOptions: ClientOptions = {
     node: endpoint,
-    ssl: {
-      requestCert: true,
-      rejectUnauthorized: true,
-    },
+    ssl: sslConfig,
     plugins: registeredSchema,
   };
 
   return clientOptions;
 }
+
+export const readCertificateAuthorities = (rawConfig: any) => {
+  let certificateAuthorities: string[] | undefined;
+
+  const addCAs = (ca: string[] | undefined) => {
```

Comment:
Nit:
For me, it's more readable to not using abbreviation.
addCAs ----> addCertificateAuthorities
Any concerns?

Reply:
I chose the abbreviation since its consistent with how the opensearch client can be configured [here](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/server/opensearch/opensearch_config.ts#L364-L425), but I can update this to not use an abbreviation. CA is the standard abbr for certificate authority

Reply:
Changed all abbreviations to the long form - certificateAuthorities.